### PR TITLE
Update performance tests

### PR DESCRIPTION
**Pull request by whynick1**
1) Update tests regarding new ducktape feature (allocate node based on [os | cpu | mem | disk | additional_disks])
2) Support performance test with multiple partitions
3) Fix bug: CLI test not running
4) Fix gradle build: Out of Memory error

### CODE REVIEW

**Comment by whynick1** [comment]
```
[standard] num_txn=100000.num_active_partitions=1.txn_size=512.timeout=360
####################### CONSUMER PERFORMANCE REPORT #######################
Read 100000 transactions, with total 48.8281 MB, in 31.08 secs
Transaction/sec: 3217.9174
MB/sec: 1.5712
###########################################################################

[double txn size] num_txn=100000.num_active_partitions=1.txn_size=1024.timeout=360
####################### CONSUMER PERFORMANCE REPORT #######################
Read 100000 transactions, with total 97.6563 MB, in 36.39 secs
Transaction/sec: 2748.3853
MB/sec: 2.6840
###########################################################################

[2 partitions] num_txn=100000.num_active_partitions=4.txn_size=512.timeout=360
####################### CONSUMER PERFORMANCE REPORT #######################
Read 100000 transactions, with total 48.8281 MB, in 19.29 secs
Transaction/sec: 5183.7644
MB/sec: 2.5311
###########################################################################

[standard] lock_pool_size=0.txn_size=512.num_active_partitions=1.interval=10.timeout=360.num_thread=100.txn_per_thread=1000
####################### PRODUCER PERFORMANCE REPORT #######################
Appended 100000 transactions, with total 48.8281 MB, in 66.49 secs
Transaction/sec: 1504.0308
MB/sec: 0.7344
Retry/Transaction: 0.0000
MilliSec/Transaction(client side congestion excluded): 304.4658
###########################################################################

[double txn size] lock_pool_size=0.txn_size=1024.num_active_partitions=1.interval=10.timeout=360.num_thread=100.txn_per_thread=1000
####################### PRODUCER PERFORMANCE REPORT #######################
Appended 100000 transactions, with total 97.6563 MB, in 64.45 secs
Transaction/sec: 1551.5904
MB/sec: 1.5152
Retry/Transaction: 0.0000
MilliSec/Transaction(client side congestion excluded): 301.8035
###########################################################################

[half clients] lock_pool_size=0.txn_size=512.num_active_partitions=1.interval=10.timeout=360.num_thread=50.txn_per_thread=2000
####################### PRODUCER PERFORMANCE REPORT #######################
Appended 100000 transactions, with total 48.8281 MB, in 49.91 secs
Transaction/sec: 2003.6466
MB/sec: 0.9783
Retry/Transaction: 0.0000
MilliSec/Transaction(client side congestion excluded): 116.1583
###########################################################################

[double interval] lock_pool_size=0.txn_size=512.num_active_partitions=1.interval=20.timeout=360.num_thread=100.txn_per_thread=1000
####################### PRODUCER PERFORMANCE REPORT #######################
Appended 100000 transactions, with total 48.8281 MB, in 68.09 secs
Transaction/sec: 1468.5797
MB/sec: 0.7171
Retry/Transaction: 0.0000
MilliSec/Transaction(client side congestion excluded): 302.9227
###########################################################################

[4 partitions] lock_pool_size=0.txn_size=512.num_active_partitions=4.interval=10.timeout=360.num_thread=100.txn_per_thread=1000
####################### PRODUCER PERFORMANCE REPORT #######################
Appended 100000 transactions, with total 48.8281 MB, in 60.24 secs
Transaction/sec: 1659.9163
MB/sec: 0.8105
Retry/Transaction: 0.0000
MilliSec/Transaction(client side congestion excluded): 666.2706
###########################################################################

[with lock] lock_pool_size=64.txn_size=512.num_active_partitions=1.interval=10.timeout=360.num_thread=100.txn_per_thread=100
####################### PRODUCER PERFORMANCE REPORT #######################
Appended 10000 transactions, with total 4.8828 MB, in 78.00 secs
Transaction/sec: 128.2133
MB/sec: 0.0626
Retry/Transaction: 2.8495
MilliSec/Transaction(client side congestion excluded): 14304.1325
###########################################################################

[double lock size] lock_pool_size=128.txn_size=512.num_active_partitions=1.interval=10.timeout=360.num_thread=100.txn_per_thread=100
####################### PRODUCER PERFORMANCE REPORT #######################
Appended 10000 transactions, with total 4.8828 MB, in 46.37 secs
Transaction/sec: 215.6567
MB/sec: 0.1053
Retry/Transaction: 1.6328
MilliSec/Transaction(client side congestion excluded): 4808.2443
###########################################################################

[double lock size; 2 partition] lock_pool_size=128.txn_size=512.num_active_partitions=2.interval=10.timeout=360.num_thread=100.txn_per_thread=100
####################### PRODUCER PERFORMANCE REPORT #######################
Appended 10000 transactions, with total 4.8828 MB, in 24.14 secs
Transaction/sec: 414.3189
MB/sec: 0.2023
Retry/Transaction: 1.4469
MilliSec/Transaction(client side congestion excluded): 2425.0235
###########################################################################
```

File: build.gradle [hunk 1]
```diff
@@ -205,19 +210,14 @@ project(':waltz-server') {
             project(':waltz-storage'),
 
             "com.fasterxml.jackson.core:jackson-databind:$jacksonVersion",
-            "com.wepay.riff:riff-config:$riffVersion",
```
Comment (criccomini) [inline build.gradle]: Curious about removal of these. Why? I believe these modules are using code from these JARs. Should not depend on transitive dependencies to pull them in.. should be defined explicitly if used in this module.

Comment (whynick1) [reply build.gradle]: > Should not depend on transitive dependencies to pull them in.. should be defined explicitly if used in this module.
My thought was `waltz-storage` inherits these dependencies from `waltz-common`. But what you say make sense to me, I will add them back!

File: waltz-tools/src/test/java/com/wepay/waltz/tools/storage/StorageCliTest.java [hunk 2]
```diff
@@ -502,12 +502,8 @@ public void testRecoverPartition() throws Exception {
             maxTransactionId = (Long) destinationClient.getMaxTransactionId(-1, partitionId).get();
             assertNotNull(maxTransactionId);
             assertEquals(9, maxTransactionId.longValue());
-
-            sourceClient.close();
```
Comment (ymatsuda) [inline waltz-tools/src/test/java/com/wepay/waltz/tools/storage/StorageCliTest.java]: Why did you remove these two lines?

Comment (whynick1) [reply waltz-tools/src/test/java/com/wepay/waltz/tools/storage/StorageCliTest.java]: Aha, good catch! Thought `helper.closeAll()` would also close `sourceClient`, but it is not true, will revert this change.
